src/kaltura-player.js

```diff
@@ -794,6 +852,99 @@ class KalturaPlayer extends FakeEventTarget {
   get crossOrigin(): ?string {
     return this._localPlayer.crossOrigin;
   }
+
+  /**
+   * Gets the player visibility state
+   * @returns {boolean} - whether the player is in the active browser tab and visible in the view port
+   * @public
+   */
+  get isVisible(): boolean {
+    return this._isVisible;
+  }
+
+  /**
+   * Gets the player viewability manager service
+   * @returns {ViewabilityManager} - player viewability manager
+   * @public
+   */
+  get viewabilityManager(): ViewabilityManager {
+    return this._viewabilityManager;
+  }
+  _initScrollVisibility() {
+    this._viewabilityManager.observe(
+      Utils.Dom.getElementById(this.config.ui.targetId),
+      this.config.visibility.playerThreshold / 100,
+      this._handleScrollVisibilityChange.bind(this)
+    );
+  }
+
+  _handleScrollVisibilityChange(visible: boolean) {
+    this._isVisibleInScroll = visible;
+    this._handleVisibilityChange();
+  }
+
+  _handleVisibilityChange() {
+    const prevVisibility = this._isVisible;
+    this._isVisible = this._isTabVisible && this._isVisibleInScroll;
+
+    if (prevVisibility !== this._isVisible) {
+      this.dispatchEvent(new FakeEvent(CoreEventType.VISIBILITY_CHANGE, {visible: this._isVisible}));
+    }
+
+    if (this.config.playback.autoplay === AutoPlayType.IN_VIEW && this._isVisible && !this._playbackStart) {
+      this._localPlayer.autoPlay();
+    }
+  }
+
+  _initTabVisibility(): void {
+    KalturaPlayer._logger.debug('_initTabVisibility');
+    if (typeof document.hidden !== 'undefined') {
+      // Opera 12.10 and Firefox 18 and later support
+      this._visibilityTabHiddenAttr = 'hidden';
+      this._visibilityTabChangeEventName = 'visibilitychange';
+    } else if (typeof document.msHidden !== 'undefined') {
+      this._visibilityTabHiddenAttr = 'msHidden';
+      this._visibilityTabChangeEventName = 'msvisibilitychange';
+    } else if (typeof document.webkitHidden !== 'undefined') {
+      this._visibilityTabHiddenAttr = 'webkitHidden';
+      this._visibilityTabChangeEventName = 'webkitvisibilitychange';
+    }
+
+    if (this._visibilityTabHiddenAttr && this._visibilityTabChangeEventName) {
+      this._eventManager.listen(document, this._visibilityTabChangeEventName, this._handleTabVisibilityChange.bind(this));
+      this._isTabVisible = !document[this._visibilityTabHiddenAttr];
+    }
+  }
```

Comment:
we can move all of this to the viweability manager, at the end of the day either scroll or tab viewability boils down to isVisible

Reply:
If so I need to change the API of viewability Manager - I need to change the name of the observe to observeInViewPort and add observeTab, or something like this. because they have different params (one needs to send an element to observe)

Reply:
done